php/EE/Model/Base.php

```diff
@@ -0,0 +1,323 @@
+	 */
+	public function offsetGet( $index ) {
+		return $this->__get( $index );
+	}
+
+	/**
+	 * Overriding offsetSet for correct behaviour while saving object properties by array index.
+	 *
+	 * @param string|int $index Name of property to check
+	 * @param mixed  $value Value of property to set
+	 *
+	 * @throws \Exception
+	 *
+	 * @return bool
+	 */
+	public function offsetSet( $index, $value ) {
+		return $this->__set( $index, $value );
+	}
+
+	/**
+	 * Overriding offsetGet for correct behaviour while checking array_key_exists.
+	 *
+	 * @param string|int $index Name of property to check
+	 *
+	 * @throws \Exception
+	 *
+	 * @return bool
+	 */
+	public function offsetExists( $index ) {
+		return $this->__isset( $index );
+	}
+
+	/**
+	 * Overriding offsetUnset for correct behaviour while deleting object properties by array index.
+	 *
+	 * @param string|int $index Name of property to check
+	 *
+	 * @throws \Exception
+	 */
+	public function offsetUnset( $index ) {
+		$this->__unset( $index );
+	}
+
+	/**
+	 * Returns single model fetched by primary key
+	 *
+	 * @param string $value value to find
+	 * @param string $column Column to find in. Defaults to primary key
+	 *
+	 * @throws \Exception
+	 *
+	 * @return static
+	 */
+	public static function find( string $value, string $column = null ) {
+		$primary_key_column = $column ?? static::$primary_key;
+		$model = EE::db()
+			->table( static::$table )
+			->where( $primary_key_column, $value )
+			->first();
+
+		if( $model === false ) {
+			return false;
+		}
+
+		return static::single_array_to_model( $model );
+	}
+
+	/**
+	 * Throws exception if model is not found
+	 *
+	 * @param string $value value to find
+	 * @param string $column Column to find in. Defaults to primary key
+	 *
+	 * @throws \Exception
+	 *
+	 * @return static
+	 */
+	public static function find_or_fail( string $value, string $column = null ) {
+		$model = static::find( $value, $column );
+
+		if ( ! $model ) {
+			throw new \Exception( sprintf( 'Unable to find %s : with primary key: %s and value: %s', __CLASS__, static::$primary_key, $value ) );
+		}
+		return $model;
+	}
+
+	/**
+	 * Exits with error if model is not found
+	 *
+	 * @param string $value value to find
+	 * @param string $column Column to find in. Defaults to primary key
+	 *
+	 * @throws \Exception
+	 *
+	 * @return static
+	 */
+	public static function find_or_error( string $value, string $column = null ) {
+		$model = static::find( $value, $column );
+
+		if ( ! $model ) {
+			EE::error( sprintf( 'Unable to find %s : with primary key: %s and value: %s', __CLASS__, static::$primary_key, $value ) );
+		}
+		return $model;
+	}
+
+	/**
+	 * Returns all models.
+	 *
+	 * @param array $columns Columns to select
+	 *
+	 * @return array Array of models
+	 * @throws \Exception
+	 */
+	public static function all( $columns = [] ) {
+		return static::many_array_to_model(
+			EE::db()
+				->table( static::$table )
+				->select( ...$columns )
+				->get()
```

Comment:
please check if you're getting some data or not then proceed further. Also you've added data type restriction already on `many_array_to_model` and this get() function can return `false` so here you'll get a warning/error for not passing `array` type data.